File: Remove_Vertex.py
def remove_vertex(vertex):
    if vertex < 1 or vertex > V[0]:
        print("Invalid vertex to remove.")
        return

    # Remove all edges associated with the vertex
    for v in range(1, V[0] + 1):
        if v == vertex:
            continue
        adj_list[v - 1] = [vtx for vtx in adj_list[v - 1] if vtx != vertex]

    # Remove the adjacency list of the removed vertex
    adj_list.pop(vertex - 1)

    # Shift vertices to fill the gap
    for v in range(vertex - 1, V[0] - 1):
        adj_list[v] = adj_list[v + 1]

    # Reduce the number of vertices
    V[0] -= 1

File: C++/Remove_Vertex.cpp
void removeVertex(int vertex) {
    if (vertex < 1 || vertex > V) {
        cerr << "Invalid vertex to remove." << endl;
        return;
    }

    // Remove all edges associated with the vertex
    for (int v = 1; v <= V; v++) {
        if (v == vertex) {
            continue;
        }
        vector<int>::iterator it = adjList[v - 1].begin();
        while (it != adjList[v - 1].end()) {
            if (*it == vertex) {
                adjList[v - 1].erase(it);
                break;
            }
            it++;
        }
    }

    // Remove the adjacency list of the removed vertex
    adjList[vertex - 1].clear();

    // Shift vertices to fill the gap
    for (int v = vertex - 1; v < V - 1; v++) {
        adjList[v] = adjList[v + 1];
    }

    // Reduce the number of vertices
    V--;
    adjList.pop_back();
}
